lib/engine/game/g_18_usa/step/buy_pullman.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../step/train'
+module Engine
+  module Game
+    module G18USA
+      module Step
+        class BuyPullman < Engine::Step::Base
+          include Engine::Step::Train
+          def actions(entity)
+            return [] if entity != current_entity
+
+            if @game.depot.upcoming.any? do |t|
```

Comment:
Order by the lowest cost check first and then work up to the highest cost check.

Reply:
Also, I think it would read more clearly if you use braces instead of a block for your lambda.

Reply:
Is can_buy_train?(entity) equivalent to can buy pullman train? We only want it to return true if so.

Reply:
Resolving the above comment obsoletes this one